### PR TITLE
Create holiday command

### PR DESCRIPTION
**Pull request by a2937**
Holiday information is cached ready for use all day.

### CODE REVIEW

File: src/main/java/com/avairebot/commands/fun/HolidayCommand.java [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+
+public class HolidayCommand extends Command
+{
+
+    private static Cache<String, List<HolidayService.Holiday>> holidayMap = CacheBuilder.newBuilder()
+        .concurrencyLevel(4) // read docs for more details
```
[the remark this answers is not in the document]
Comment (a2937) [reply src/main/java/com/avairebot/commands/fun/HolidayCommand.java]: It's there because I forgot to actually check the default value.

File: src/main/java/com/avairebot/commands/fun/HolidayCommand.java [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+     */
+    @Override
+    public boolean onCommand(CommandMessage context, String[] args)
+    {
+        Request request = RequestFactory.makeGET("https://holidayapi.com/v1/holidays");
+        LocalDate date = LocalDate.now();
+        int dayOfMonth = date.getDayOfMonth();
+        int monthOfYear = date.getMonthValue();
+        String holidayApiClientToken = avaire.getConfig().getString("apiKeys.holidayapi", "invalid");
+        if (holidayApiClientToken == null || holidayApiClientToken.length() == 32)
```
[the remark this answers is not in the document]
Comment (a2937) [reply src/main/java/com/avairebot/commands/fun/HolidayCommand.java]: Yeah I see the issue. Thanks for pointing that out.

File: src/main/resources/config.yml [hunk 3]
```diff
@@ -574,6 +574,14 @@ vote-lock:
...
+    # in order to retrieve current holiday information.
+    #
+    # You can get a free API key that allows 1000 requests per-month by singing up on
+    # their website, then going to the "API Keys" tab and creating a new key.
+    # Get a key at: https://holidayapi.com/
+    #
+    holidayapi: ''
   # Google API key, this is used for music requests to YouTube and the
   # Shorten API for shortening links, note the play command will still
   # work with youtube links without an API key.
```
Comment (Adyne) [inline src/main/resources/config.yml]: Might want to fix indenting to match the ones below, as well as a line break after `holidayapi: ''`

File: src/main/resources/langs/en_US.yml [hunk 4]
```diff
@@ -471,6 +471,13 @@ fun:
             - 'Outlook not so good'
             - 'Very doubtful'
 
+    HolidayCommand:
+        tooManyAttempts: "The Holiday API was hit too many times which shouldn't have been possible. D:"
+        notFound: "The Holiday API command could not retrieve any data for this request."
+        somethingWentWrong: "Something went wrong. Perhaps contact a bot administrator for help."
+        noHolidaysToday: "No holidays today...get back to work!"
```
Comment (Adyne) [inline src/main/resources/langs/en_US.yml]:
```suggestion
        noHolidaysToday: "No holidays today... Get back to work!"
```

**Comment by chaNcharge** [comment]
Gonna barge in here and add a few words of advice for future PRs:

* You should add a summary of what the command does and describe what you changed, in 1-2 paragraphs. 
* Use a good editor/IDE that has support for editorconfig files so we don't have to constantly point out spacing issues
* Always read and double check your code before sending in a commit or PR; Any single change you make should be tested. The last thing you want is sensitive data being shown in a commit because it's annoying to remove once its committed. 
* Get a linter or an IDE with a linter, should help with quality checking your code
* Keep code simple, short, and precise. If you feel code can be a little shorter or more efficient, it probably can be.

File: src/main/resources/langs/hu_HU.yml [hunk 5]
```diff
@@ -269,14 +269,6 @@ administration:
...
         message: "The modlog case with an ID of **:case** has been been pardoned successfully."
 
-    ModlogPardonCommand:
-        modlogNotEnabled: "No modlog channel has been set, you must set a modlog channel to use this command."
-        invalidCaseId: "Invalid case id given, the ID must be greater than 0 and less than {0}"
-        couldntFindCaseWithId: "Couldn't find a modlog case with an ID of {0} where you were the moderator, are you sure you're the moderator for the given modlog case?"
-        caseAlreadyPardoned: "The modlog case ID with an ID of `{0}` is already pardoned."
-        notAPunishment: "The modlog case ID with an ID of `{0}` is not a punishment!"
-        message: "The modlog case with an ID of **:case** has been been pardoned successfully."
-
```
Comment (Senither) [inline src/main/resources/langs/hu_HU.yml]: Why was this removed?

Comment (a2937) [reply src/main/resources/langs/hu_HU.yml]: It was a duplicate Yaml key I noticed when double-checking my pull request.
